presto-manager-agent/src/main/java/com/teradata/prestomanager/agent/PrestoInstaller.java

```diff
@@ -85,21 +95,129 @@ private static void installUsingRpm(String pathToRpm, boolean checkDependencies)
         if (installRpm != 0) {
             throw new PrestoManagerException("Failed to install Presto", installRpm);
         }
-        addTpchCatalog();
+        addTpchCatalog(CATALOG_DIR);
         LOGGER.debug("Successfully installed Presto");
     }
 
-    private static void addTpchCatalog()
+    private static void addTpchCatalog(Path directory)
             throws PrestoManagerException
     {
         try {
-            if (!isDirectory(CATALOG_DIR)) {
-                createDirectory(CATALOG_DIR);
-                write(CATALOG_DIR.resolve("tpch.properties"), "connector.name=tpch".getBytes());
+            if (!isDirectory(directory)) {
+                createDir(directory);
             }
+            write(directory.resolve("tpch.properties"), "connector.name=tpch".getBytes());
         }
         catch (IOException e) {
             throw new PrestoManagerException("Failed to add tpch catalog", e);
         }
     }
+
+    public static void installUsingTar(URL urlToFetchPackage)
+            throws PrestoManagerException
+    {
+        preInstallChecks();
+        File tarTempFile = fetchFileFromUrl(urlToFetchPackage);
+        try {
+            if (isDirectory(TAR_INSTALLATION_DIR)) {
+                deleteDirContents(TAR_INSTALLATION_DIR);
+            }
+            else {
+                createDir(TAR_INSTALLATION_DIR);
+            }
+            untarFile(tarTempFile, TAR_INSTALLATION_DIR);
+        }
+        finally {
+            if (!tarTempFile.delete()) {
+                LOGGER.warn("Failed to delete the temporary file: %s", tarTempFile.toString());
+            }
+        }
+        createDir(TAR_INSTALLATION_DIR.resolve("etc"));
```

Comment:
The `etc` directory should be stored as a separate constant; if we want to base it off of the install directory, that can be done in the configuration class. Basically, any path you create with `Path::resolve` to which my comment on `addNodeProperties` does not apply should have its own constant.